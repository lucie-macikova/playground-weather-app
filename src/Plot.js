/* global Plotly */
import React, { Component } from 'react';

export default class Plot extends Component {
	drawPlot(){
		Plotly.newPlot('plot', [{
			x: this.props.xData.toJS(),
			y: this.props.yData.toJS(),
			type: this.props.type
		}])
		this.plot.on('plotly_click', this.props.onPlotClick)

	}
	componentDidMount(){
		this.drawPlot()
	}
	componentDidUpdate(){
		this.drawPlot()
	}
	shouldComponentUpdate (nextProps){
		const xDataChanged = !this.props.xData.equals(nextProps.xData)
		const yDataChanged = !this.props.yData.equals(nextProps.yData)

		return xDataChanged || yDataChanged
	}

	render(){
		console.log('rendering')
		return <div id="plot" ref={node => this.plot = node} ></div>
	}
}
