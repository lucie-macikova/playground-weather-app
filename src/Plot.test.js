import React from 'react';
import renderer from 'react-test-renderer'
import Plot from './Plot';
import {fromJS} from 'immutable'

function createNodeMock(element){
  if(element.props.id === "plot"){
    return {
      on: () => {}
    }
  }
}
it('renders without crashing', () => {
  global.Plotly = {
    newPlot: () => {}
  }
  const options ={createNodeMock}
  const tree = renderer.create(<Plot xData={fromJS({})} yData={fromJS({})} />, options).toJSON()
  expect(tree).toMatchSnapshot()
});
