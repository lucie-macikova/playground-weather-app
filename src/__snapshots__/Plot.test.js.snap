exports[`test renders without crashing 1`] = `
<div
  id="plot" />
`;
